test/extensions/filters/http/common/compressor/compressor_filter_test.cc

```diff
@@ -22,8 +22,8 @@ namespace HttpFilters {
 namespace Common {
 namespace Compressors {
 
-class MockCompressor : public Compressor::Compressor {
-  void compress(Buffer::Instance&, ::Envoy::Compressor::State) override {}
+class MockCompressor : public Envoy::Compression::Compressor::Compressor {
```

Comment:
This could be an actual mock, and the class below could be skipped. In addition per my comment above, I think we can clean the tests here that expose private functions.

Take a look at https://github.com/junr03/envoy/pull/1/files#diff-579860a53a9e1874e312dd0f7b2df943

Reply:
I find it convenient to test private functions in isolation from each other. There's a TODO item though to move them in a separate class (using the pimpl idiom I presume).

Reply:
nice, yeah I agree some of the private functions (for instance the ones that manipulate headers) could become functions in a static utility class that is shared between compressor and decompressor. We can do that in a subsequent PR.